src/pudl/analysis/allocate_net_gen.py

```diff
@@ -514,24 +527,36 @@ def associate_generator_tables(
 ) -> pd.DataFrame:
     """Associate the three tables needed to assign net gen and fuel to generators.
 
-    The ``generation_fuel_eia923`` table's data is reported at the
-    :py:const:`IDX_PM_ESC` granularity. Each generator has one ``prime_mover_code``, but
-    potentially several ``energy_source_code``s (reported in as primary source,
-    secondary source, etc.). Much of the data in ``generation_fuel_eia923`` not
-    associated with primary energy sources, so before merging the data with the
-    generators, we employ :func:``stack_generators``.
-
-    After the generator's energy source codes have been stacked, we group the other
-    data tables and merge them with the generators. We add suffixes to the data columns
-    to identify the source table.
+    The :ref:`generation_fuel_eia923` table's data is reported at the
+    :py:const:`IDX_PM_ESC` granularity. Each generator in the :ref:`generators_eia860`
+    has one ``prime_mover_code``, but potentially several ``energy_source_code``s that
+    are reported in several columns. We need to reshape the generators energy source
```

Comment:
"We need to reshape the generators table such that each generator has a separate record corresponding to each of its reported `energy_source_code`s, so it can be merged with the..."